File: app/api/v1/graph.py
from fastapi import APIRouter, Depends
from pydantic import BaseModel

from app.core.response import success
from app.services.graph_service import get_graph_service
from app.schemas.common import GraphQueryRequest
from app.core.deps import CurrentUser, require_roles

router = APIRouter()

# class SymptomCheckRequest(BaseModel):
#     symptoms: list[str]



# @router.get("/symptoms")
# def get_symptoms():
#     """获取所有症状列表"""
#     data = get_graph_service().get_all_symptoms()
#     return success(data)


@router.post("/infer")
def infer_diseases(req: GraphQueryRequest, _: CurrentUser = Depends(require_roles("user", "admin", "doctor"))):
    """根据症状推理可能疾病"""
    results = get_graph_service().infer_diseases_by_symptoms(req.symptoms)
    return success(results)


@router.get("/disease/{name}")
def disease_detail(name: str):
    """获取疾病详情"""
    detail = get_graph_service().get_disease_detail(name)
    return success(detail)


@router.get("/full")
def full_graph():
    """获取完整知识图谱（默认可视化）"""
    graph = get_graph_service().get_full_graph()
    return success(graph)

@router.get("/subgraph")
def entity_subgraph(entity: str):
    """获取实体子图（可视化）"""
    graph = get_graph_service().get_entity_subgraph(entity)
    return success(graph)


@router.get("/search")
def search_entities(keyword: str):
    """搜索图谱实体"""
    results = get_graph_service().search_entities(keyword)
    return success(results)


@router.get("/stats")
def graph_stats(_: CurrentUser = Depends(require_roles("admin"))):
    """图谱统计"""
    stats = get_graph_service().get_graph_stats()
    return success(stats)

File: frontend/src/api/graph.js
import request from './request'

// 症状推理疾病
export function inferDisease(symptoms) {
  return request({
    url: '/graph/infer',
    method: 'post',
    data: { symptoms }
  })
}

// 疾病详情
export function getDiseaseDetail(name) {
  return request({
    url: `/graph/disease/${encodeURIComponent(name)}`,
    method: 'get'
  })
}

// 完整知识图谱
export function getFullGraph() {
  return request({
    url: '/graph/full',
    method: 'get'
  })
}

// 实体子图
export function getSubgraph(entity) {
  return request({
    url: '/graph/subgraph',
    method: 'get',
    params: { entity }
  })
}

// 搜索图谱实体
export function searchGraph(keyword) {
  return request({
    url: '/graph/search',
    method: 'get',
    params: { keyword }
  })
}

// 图谱统计
export function getGraphStats() {
  return request({
    url: '/graph/stats',
    method: 'get'
  })
}
